libs/product/driver/magento/src/transforms/product-preview.ts

```diff
@@ -5,21 +5,39 @@ import {
   DaffProductDiscount,
 } from '@daffodil/product';
 
+import { MagentoBundledProduct } from '../models/bundled-product';
+import { MagentoConfigurableProduct } from '../models/configurable-product';
 import {
   MagentoProduct,
   MagentoProductTypeEnum,
 } from '../models/magento-product';
 import { MagentoProductPreview } from '../models/product-preview.interface';
+import { transformMagentoBundledProduct } from './bundled-product-transformers';
+import { transformMagentoConfigurableProduct } from './configurable-product-transformers';
 
-const productTypeMap = {
-  [MagentoProductTypeEnum.BundledProduct]: DaffProductTypeEnum.Composite,
-  [MagentoProductTypeEnum.ConfigurableProduct]: DaffProductTypeEnum.Configurable,
-  [MagentoProductTypeEnum.SimpleProduct]: DaffProductTypeEnum.Simple,
-};
+/**
+ * Transforms a Magento product into a product preview.
+ * Handles all product types.
+ */
+export function transformMagentoProductPreview(product: MagentoProduct, mediaUrl: string): DaffProduct {
+  const preview = transformMagentoSimpleProductPreview(product, mediaUrl);
```

Comment:
I think it's technically better to just call `transformMagentoSimpleProductPreview` in each switch case, since every `transformMagentoProductPreview` call only needs to call `transformMagentoSimpleProductPreview` one time. That way, it doesn't need to save the transformed preview in memory. 

Same for the `transformMagentoProduct` function

Reply:
changed